src/DBus.vala

```diff
@@ -107,18 +128,45 @@ public class Notifications.Server : Object {
             id
         );
         bubble.show_all ();
+    }
 
-        send_sound (hints);
+    private void send_confirmation (string icon_name, HashTable<string, Variant> hints) {
+        double progress_value;
+        if (hints.contains ("value")) {
+            progress_value = hints.@get ("value").get_int32 ().clamp (0, 100) / 100.0;
```

Comment:
Should probably use `hints.lookup` like in `send_sound`: `Variant? variant = hints.lookup ("category");`.

Reply:
oops, admittedly copy/pasted that. My bad!